src/xrCore/xr_ini.h

```diff
@@ -69,195 +56,235 @@ class XRCORE_API CInifile
     string_path m_file_name;
     Root DATA;
 
-    void Load(IReader* F, pcstr path, allow_include_func_t allow_include_func = nullptr);
+    void Load(IReader* reader, pcstr path, allow_include_func_t allow_include_func, u8 include_depth);
+
+    bool insert_new_section(Sect* section);
+
+    [[nodiscard]] Sect& unchecked_read_section(pcstr section_name);
+    [[nodiscard]] const Sect& unchecked_read_section(pcstr section_name) const;
 
 public:
-    CInifile(IReader* F, pcstr path = nullptr, allow_include_func_t allow_include_func = nullptr);
+    CInifile(IReader* reader, pcstr path = nullptr, allow_include_func_t allow_include_func = nullptr);
 
     CInifile(pcstr fileName, bool readOnly = true,
              bool loadAtStart = true, bool saveAtEnd = true,
              u32 sect_count = 0, allow_include_func_t allow_include_func = nullptr);
 
-    virtual ~CInifile();
+    ~CInifile();
```

Comment:
The class is being inherited as base for CScriptIniFile class, so it's better to leave virtual destructor here.

Reply:
Ups